lib/howl/interact.moon

```diff
@@ -0,0 +1,29 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+import dispatch from howl
+
+interactions = {}  -- registry
+
+register = (spec) ->
+  for field in *{'name', 'description'}
+    error 'Missing field for command: "' .. field .. '"' if not spec[field]
+
+  if not (spec.factory or spec.handler) or (spec.factory and spec.handler)
+    error 'One of "factory" or "handler" required'
+
+  interactions[spec.name] = moon.copy spec
+
+unregister = (name) ->
+  interaction = interactions[name]
+  return if not interaction
```

Comment:
The extra check doesn't seem to have a purpose here, we could just assign `nil` directly it seems.